scrunch/datasets.py

```diff
@@ -1441,12 +1480,32 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_tuple, dataset):
+        self.shoji_tuple = var_tuple
+        self.is_instance = False
+        self._resource = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
+            self.dataset = dataset
+        else:
+            # Need to instantiate Dataset for ds.order method
+            self.dataset = Dataset(dataset)
+
+    @property
+    def resource(self):
+        if not self.is_instance:
+            self._resource = self.shoji_tuple.entity
+            self.is_instance = True
+        return self._resource
+
+    @property
+    def url(self):
+        return self.resource.self
```

Comment:
You don't need to have the resource hydrated to get the url. You can set the `self.url` on init by just getting it from `tuple.entity_url` and save yourself a GET request unnecessarily here.